### PR TITLE
Change farming mode to move all items not tagged or locked

### PR DESCRIPTION
**Pull request by imeredith**
Opened this PR instead of https://github.com/DestinyItemManager/DIM/pull/4817

I really happy with the functionality. The only thing that worries me is that its a massive change to how it works and users probably will be WTFing.

### CODE REVIEW

**Comment by bhollis** [comment]
Nice! @sundevour, let me know how much this will cause people to pull out their pitchforks?

**Comment by imeredith** [comment]
Of course currently farming mode is barely working so I'm sure they would get over it :D

**Comment by sundevour** [comment]
no objections to the intended functionality, i like it. i will test it out after reset.

*as explained*, sounds like it _maintains space_.
however, i do want to make sure it doesn't lose part of what farming mode does: *make space*.
switching it on should always create a minimum 1 space.

***

because of the change, the "per" in the explanation can go.

the absolute clearest verbiage, unmisinterpretable, would be either, casually,
"DIM is preventing items from going to the Postmaster by making sure there's always empty space in each equipment slot on {{store}}. It will move unlocked or untagged items to the vault."

or to emphasize the corresponding DIM concepts,
"move items to the vault unless they are Locked or Tagged"

**Comment by sundevour** [comment]
this is trying to force all my shaders into the vault

**Comment by imeredith** [comment]
ugh yeah. I think if we just keep one slot open it would be fine. Totally
forgot about shaders

File: src/app/farming/d2farming.service.ts [hunk 1]
```diff
@@ -91,48 +92,35 @@ class D2Farming {
...
-          if (i.primStat) {
-            value += i.primStat.value / 1000;
-          }
-          return value;
-        }
-      );
-      if (itemToMove) {
-        itemsToMove.push(itemToMove);
+    // Shaders are a special case. Only free up one space.
+    if (makeRoomBucket.hash === 2973005342) {
```
Comment (bhollis) [inline src/app/farming/d2farming.service.ts]: This probably will go nuts for consumables as well. Maybe have it be conditional on the type of bucket - for example, only move one or two from `accountWide` buckets.

File: src/app/farming/d2farming.service.ts [hunk 2]
```diff
@@ -91,48 +92,35 @@ class D2Farming {
...
+      if (items.length >= makeRoomBucket.capacity) {
+        itemsToMove.push(items[0]);
```
Comment (bhollis) [inline src/app/farming/d2farming.service.ts]: You'll want to follow the same logic we had before. For example, don't attempt to move non-transferable (`notransfer`) items.

Comment (imeredith) [reply src/app/farming/d2farming.service.ts]: Ah thanks. But not the legendary/blue proirty stuff right? It shouldnt matter about the rarity as I think think that really affects what people like

File: src/app/farming/d2farming.service.ts (hunk 1 above)
Comment (imeredith) [reply src/app/farming/d2farming.service.ts]: will do

File: src/app/farming/d2farming.service.ts (hunk 2 above)
Comment (bhollis) [reply src/app/farming/d2farming.service.ts]: That shouldn't matter, but you'll need to pre-filter candidates.

**Comment by imeredith** [comment]
Sorry this is taking so long, im on vacation and arent regularly coding.

**Pull request closed without merging**